File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
    map<int,int> freq ; int majority =0 ; int frequancy =0;
    for (int i = 0; i < nums.size(); ++i)
        freq[nums[i]]++;
    for (auto it =freq.begin()  ; it != freq.end() ; ++it) {
        if (it->second>frequancy){
            majority=it->first;
            frequancy =it->second;
        }
    }
    return majority;
}
};

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
     public int MajorityElement(int[] nums)
{
    IDictionary<int,int> map = new Dictionary<int,int>();
    foreach(int num in nums)
    {
        if (map.ContainsKey(num)) map[num]++;
        else map.Add(num, 1);
    }
        
    foreach(var num in map)
    {
        if (num.Value > nums.Length / 2)
            return num.Key;
    }
    return 0;
}
        }
